src/Microsoft.Azure.WebJobs.Host/Singleton/IDistributedLockManager.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Manage distributed locks. A lock is specified by (account, lockId). 
+    /// </summary>
+    /// <remarks>
+    /// The default implementation of this is based on blob leases. 
+    /// 1. Account can be null or it may be a storage account name intended for <see cref="IStorageAccountProvider"/>. 
+    /// 2. lockId has the same naming restrictions as blobs. 
+    /// 3. For distributed locks, this implementation is responsible for renewing the locks and setting <see cref="IDistributedLock.LeaseLost"/> if the lock can't be renewed. 
```

Comment:
+1. It would be good to just expect the implementation to renew the lock when told to do so as per an appropriate renewal policy that is outside the scope of the specific IDistributedLockManager implementation. Something closer to https://github.com/shrishrirang/azure-webjobs-sdk/blob/builddev/src/Microsoft.Azure.WebJobs.Host/Lease/SqlLeaseProxy.cs#L103 would be good. In this implementation the Sql implementation just renews the lease when told to do so.